check/src/unify_type.rs

```diff
@@ -133,11 +133,8 @@ impl<I> Substitutable for ArcType<I> {
     type Variable = TypeVariable;
 
     fn new(id: u32) -> ArcType<I> {
-        Type::variable(TypeVariable::new(id))
-    }
-
-    fn from_variable(var: TypeVariable) -> ArcType<I> {
-        Type::variable(var)
+        // FIXME: using `Kind::hole()` here results in test failures
+        Type::variable(TypeVariable::new(id, Kind::typ()))
```

Comment:
The bane of premature abstraction 😆 

I would like to have a look at how other type checkers do it... full disclosure - been experimenting with implementing a separate System-Fω checker in a local repo to learn how it works 😅